gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcHeaderInterceptor.java

```diff
@@ -58,7 +57,11 @@ public GrpcHeaderInterceptor(String header) {
     return new SimpleForwardingClientCall<ReqT, RespT>(call) {
       @Override
       public void start(ClientCall.Listener<RespT> responseListener, Metadata headers) {
-        headers.put(HEADER_KEY, header);
+        for (Map.Entry<String, String> header : GrpcHeaderInterceptor.this.headers.entrySet()) {
+          Metadata.Key<String> headerKey =
```

Comment:
This is static data, and probably should be placed in the headers map itself (ideally change:
`GrpcHeaderInterceptor.this.headers` from `Map<String, String>` to `Map<Metadata.Key, String>`, so here we can just do `for(header : GrpcHeaderInterceptor.this.entrySet()) {headers.put(header.getKey(), header.getValue())}`